### PR TITLE
feat(gui): add reporting plugin for cpu temperature(s)

### PR DESCRIPTION
**Pull request by dnandha**
Closed #316 because of Jenkins checkout error
Re-Opened with shorter branch name

I'm adding a plugin to the gui reporting page which displays a graph for cpu core temperatures. The cpu temperature data is already provided as a set of rrd files by the freenas os - so why not visualize these, right?

### CODE REVIEW

**Comment by dnandha** [comment]
Jenkins: retest this please

**Comment by dnandha** [comment]
Jenkins retest this please

**Comment by kmoore134** [comment]
Our jenkins builder appears to be having issues. I'll take a look in the me and get this fixed. Hang tight!

**Comment by kmoore134** [comment]
Jenkins: retest this please

**Comment by kmoore134** [comment]
Jenkins: retest this please

**Comment by william-gr** [comment]
Also, this might render very poorly on big machines with dozens of CPUs, perhaps there is a way to make one graph per processor (package)?

**Comment by dnandha** [comment]
@william-gr 
on your first remark: that's actually something which I have been thinking how to solve it. Yes, if there are no temperature rrd files the graph would not render and a "image not found" like box appears. I'm thinking about introducing a function like "is_plugin_available" to check whether the plugin is supported on the machine.
on your last remark: to my knowledge the most cores we can get is currently 16 with AMD threadripper. I'm not sure if the graph is able to render 16 lines properly. However, I would prefer this over having 16 separate graphs for reasons of clarity (not having to scroll).

**Comment by william-gr** [comment]
@dnandha 
On first:this can also probably be accomplished by implementing `get_identifiers` and returning an empty list.

On second: Agree, but maybe you did not understand my point. I am thinking of big servers, on which you can have multiple processors. All cores of all processors would be rendered in the same graph as it currently is.

e.g.
FreeBSD/SMP: Multiprocessor System Detected: 8 CPUs
FreeBSD/SMP: 2 package(s) x 4 core(s)

# sysctl kern.smp.cpus
kern.smp.cpus: 8


But perhaps this is fine as a first step.

**Comment by dnandha** [comment]
@william-gr 
on first: get_indentifiers will naturally return None if not overridden (when we only have one identifier), so that's no option. Please check my last commit for my suggestion.
on second: ok, i got your point now. need to think about it though..

File: gui/reporting/rrd.py [hunk 1]
```diff
@@ -250,6 +251,54 @@ def graph(self):
...
+
+    def __init__(self, base_path, identifier=None, unit=None, step=None):
+        super(CPUTempPlugin, self).__init__(base_path, identifier, unit, step)
+                
+        proc = pipeopen('/sbin/sysctl -n kern.smp.cpus', important=False, logger=log)
+        proc_out = proc.communicate()[0]
+        try:
+            self.cpu_count = int(proc_out)
+        except:
+            self.is_available = False
```
Comment (william-gr) [inline gui/reporting/rrd.py]: I dont understand how having a kern.smp.cpus sysctl is the key for it having the CPU Temperature report working or not.

**Comment by william-gr** [comment]
@dnandha 
first: I believe you're wrong there. None != []. If you implement get_identifiers returning an empty list the graph will not show, try it.

**Comment by dnandha** [comment]
@william-gr aah ok, empty list, not None. ok will try that

**Comment by william-gr** [comment]
Merged, thanks!

**Pull request merged**